lib/src/adapter_interface.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:http_mock_adapter/src/handlers/request_handler.dart';
+import 'package:http_mock_adapter/src/request.dart';
+
+/// Top level interface of all the adapters.
+/// It is implemented by [MainInterceptor] and [DioAdapter].
+///
+/// This Interface is use for 2 reqsons:
+/// 1. To maintain the nice and best practice orinted way of architecture
+/// 2.  Because of the reason that [RequestHandler] is returning different
+/// class instances form [reply] method, depending on its generic type
+/// parameters by using [dynamic] type for [reply] was removing
+/// autocomplition ability, that's why both Adapters implement this interface
+/// to provide good developer experience.
+abstract class AdapterInterface {
```

Comment:
I think this is unnecessary as `reply()` method can return an instance of type **T** of which either `DioAdapter` and `MainInterceptor` be subclasses of by comparing the types at runtime.

For demonstration purposes:

```dart
T reply(int statusCode, dynamic data) {
    ...

    for (Object adapter in [DioAdapter(), MainInterceptor()]) {
      if (adapter.runtimeType == T) {
        return adapter as T;
      }
    }

  throw RequestHandlerException();
}
```

We should either keep this class and remove `RequestRouted` mixin, or remove this class and keep `RequestRouted` mixin.

If not, a lot of code will be duplicated without necessity.

Also, the documentation could use some improvement.

Reply:
Really cool way.

Reply:
editor code completion is disabled with this method we need, Interface for sure, but another method.

Reply:
Where (or from what) specifically is the editor code completion disabled?

Reply:
when you are typing `intercept.onGet().Reply()` after `Reply` you don't get the list of `onget,onpost,onput....` just getting `toString`  and `hashcode`,

Reply:
Understood, thank you. That's definitely a valid concern.

I will look into alternatives and get back to you soon.

Reply:
როდის იქნები სლაკს ?

Reply:
ვარ უკვე.